demos/TicTacToe/Classes/MainMenuScene.cpp

```diff
@@ -22,33 +76,345 @@ bool MainMenuScene::init() {
+  Director::getInstance()
+      ->getEventDispatcher()
+      ->addEventListenerWithSceneGraphPriority(anonymous_label_touch_listener,
+                                               anonymous_login_label);
+  auth_background->addChild(anonymous_login_label);
+
+  // Extracting the origin, size and position of the text field so that the
+  // border can be created based on those values.
+  auto email_text_field_origin = Vec2(0, 0);
+  auto email_text_field_position = Size(110, 350);
+  auto text_field_padding = 20;
+  auto email_font_size = 36.0;
+  auto email_text_field_size = Size(400, 2 * email_font_size);
+
+  // Setting up the constraints of the border so it surrounds the text box.
+  Vec2 email_border_corners[4] = {
+      Vec2(email_text_field_position.width - text_field_padding,
+           email_text_field_position.height),
+      Vec2(email_text_field_position.width + email_text_field_size.width,
+           email_text_field_position.height),
+      Vec2(email_text_field_position.width + email_text_field_size.width,
+           email_text_field_position.height + email_text_field_size.height),
+      Vec2(email_text_field_position.width - text_field_padding,
+           email_text_field_position.height + email_text_field_size.height)};
+
+  // Creating a text field border and adding it around the text field.
+  auto email_text_field_border = DrawNode::create();
+  email_text_field_border->drawPolygon(email_border_corners, 4,
+                                       Color4F(0, 0, 0, 0), 1, Color4F::WHITE);
+  // Creating a text field to enter the user's email.
+  email_text_field = cocos2d::TextFieldTTF::textFieldWithPlaceHolder(
+      "enter an email address", email_text_field_size, TextHAlignment::LEFT,
+      "Arial", email_font_size);
+  email_text_field->setPosition(email_text_field_position);
+  email_text_field->setAnchorPoint(Vec2(0, 0));
+  email_text_field->setColorSpaceHolder(Color3B::GRAY);
+  email_text_field->setDelegate(this);
+
+  auto email_text_field_touch_listener = EventListenerTouchOneByOne::create();
+
+  email_text_field_touch_listener->onTouchBegan =
+      [this](cocos2d::Touch* touch, cocos2d::Event* event) -> bool {
+    if (previous_state != current_state || current_state != kAuthState)
+      return true;
+    auto bounds = event->getCurrentTarget()->getBoundingBox();
+    auto point = touch->getLocation();
+    if (bounds.containsPoint(point)) {
+      // Show the on screen keyboard and places character inputs into the text
+      // field.
+      auto str = email_text_field->getString();
+      auto text_field = dynamic_cast<TextFieldTTF*>(event->getCurrentTarget());
+      text_field->setCursorEnabled(true);
+      text_field->attachWithIME();
+    } else {
+      auto text_field = dynamic_cast<TextFieldTTF*>(event->getCurrentTarget());
+      text_field->setCursorEnabled(false);
+      text_field->detachWithIME();
+    }
+
+    return true;
+  };
+
+  // Attaching the touch listener to the text field.
```

Comment:
I left a bunch of these comments. I generally think that being-verbs shouldn't be in comments. Instead pretend you're instructing someone to do.  "Attach" instead of "Attaching", "Construct" instead of "Constructing", "Extract" instead of "Extracting", etc.